pkg/logs/sender/destination_sender.go

```diff
@@ -69,6 +77,32 @@ func (d *DestinationSender) Stop() {
 	close(d.retryReader)
 }
 
+func (d *DestinationSender) canSend() bool {
+	if d.destination.IsHA() {
+		if !d.sendEnabled {
+			if d.config.GetBool("ha.enabled") && d.config.GetBool("ha.failover") {
+				d.haLock.Lock()
+				d.sendEnabled = true
+				d.haLock.Unlock()
+
+				log.Infof("Forwarder for domain %v has been failed over to, enabling it for HA.", d.destination.Target())
+			} else {
+				log.Debugf("Forwarder for domain %v is disabled; dropping transaction for this domain.", d.destination.Target())
+			}
+		} else {
+			if !d.config.GetBool("ha.enabled") || !d.config.GetBool("ha.failover") {
+				d.haLock.Lock()
+				d.sendEnabled = false
```

Comment:
Honestly, I don't know what the concurrent accesses would or wouldn't be, so I erred on the side of simply protecting a known mutation (`sendEnabled`) rather than play whackamole.

Reply:
Since `canSend` is only called from `Send` I think it's safe to remove the lock. `Send` should always be called from the same thread. 
Could also use an `atomic.Bool` if there is a real concern that I am missing (and to save a few LOC).

Reply:
Sure, I can remove it.

Reply:
Addressed in a631176f5c.